diff --git a/libs/gui/Android.mk b/libs/gui/Android.mk
index 050b933..3830d90 100644
--- a/libs/gui/Android.mk
+++ b/libs/gui/Android.mk
@@ -30,10 +30,12 @@ LOCAL_SHARED_LIBRARIES := \
 	libGLESv2
 
 ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
+ifneq ($(BOARD_USES_LEGACY_QCOM),true)
 LOCAL_SHARED_LIBRARIES += \
         libQcomUI
 LOCAL_C_INCLUDES := hardware/qcom/display/libqcomui
 LOCAL_CFLAGS += -DQCOM_HARDWARE
+endif
 ifeq ($(TARGET_QCOM_HDMI_OUT),true)
 LOCAL_CFLAGS += -DQCOM_HDMI_OUT
 endif
diff --git a/libs/ui/Android.mk b/libs/ui/Android.mk
index b4c4817..6ee5dfb 100644
--- a/libs/ui/Android.mk
+++ b/libs/ui/Android.mk
@@ -68,8 +68,10 @@ LOCAL_C_INCLUDES := \
     external/skia/include/core
 
 ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
+ifneq ($(BOARD_USES_LEGACY_QCOM),true)
 LOCAL_CFLAGS += -DQCOM_HARDWARE
 endif
+endif
 
 LOCAL_MODULE:= libui
 
diff --git a/services/surfaceflinger/Android.mk b/services/surfaceflinger/Android.mk
index 4942b58..d7c61f9 100644
--- a/services/surfaceflinger/Android.mk
+++ b/services/surfaceflinger/Android.mk
@@ -56,10 +56,12 @@ LOCAL_C_INCLUDES := \
 LOCAL_C_INCLUDES += hardware/libhardware/modules/gralloc
 
 ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
+ifneq ($(BOARD_USES_LEGACY_QCOM),true)
 LOCAL_SHARED_LIBRARIES += \
 	libQcomUI
 LOCAL_C_INCLUDES += hardware/qcom/display/libqcomui
 LOCAL_CFLAGS += -DQCOM_HARDWARE
+endif
 ifeq ($(TARGET_QCOM_HDMI_OUT),true)
 LOCAL_CFLAGS += -DQCOM_HDMI_OUT
 endif
